sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -454,7 +459,10 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
   }
 
   // converting seconds to us
-  private[this] def longToTimestamp(t: Long): Long = SECONDS.toMicros(t)
+  private[this] def longToTimestamp(t: Long): Long = {
+    if (SQLConf.get.numericConvertToTimestampInSeconds) t * MICROS_PER_SECOND
```

Comment:
Don't think checking the flag per every values is good idea. This will badly impact on performance.